datafusion/core/Cargo.toml

```diff
@@ -84,6 +84,7 @@ parquet = { workspace = true }
 percent-encoding = "2.2.0"
 pin-project-lite = "^0.2.7"
 rand = "0.8"
+regex = "1.5.4"
```

Comment:
since this is only used for tests, I think it can be put in the `dev-dependencies` section